std/container/slist.d

```diff
@@ -135,6 +135,48 @@ struct SList(T)
         return n;
     }
 
+    private static auto createNodeChain(Stuff)(Stuff stuff)
+    if (isImplicitlyConvertible!(Stuff, T))
+    {
+        import std.range : only;
+        return createNodeChain(only(stuff));
+    }
+
+    private static auto createNodeChain(Stuff)(Stuff stuff)
+    if (isInputRange!Stuff && isImplicitlyConvertible!(ElementType!Stuff, T))
+    {
+        static struct Chain
+        {
+            Node* first;
+            Node* last;
+            size_t length;
+        }
+
+        Chain ch;
+
+        foreach (item; stuff)
+        {
+            auto newNode = new Node(null, item);
+            (ch.first ? ch.last._next : ch.first) = newNode;
+            ch.last = newNode;
+            ++ch.length;
+        }
+
+        return ch;
+    }
+
+    private static size_t insertAfterNode(Stuff)(Node* n, Stuff stuff)
+    {
+        auto ch = createNodeChain(stuff);
+
+        if (!ch.length) return 0;
+
+        ch.last._next = n._next;
```

Comment:
Doesn't this line not do anything useful, as `ch` goes out of scope at the end of the function?

Reply:
`ch` itself is going out of scope indeed, but this line is useful when inserting the new chain of nodes anywhere but the end of the list. Without this line we'd lose the tail of the list.